### PR TITLE
Add notes Closes #20, #28, #40

### PR DESCRIPTION
**Pull request by oorion**
Add notes table
Add simplecov
Add ability for admin to add notes
Add integration test

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -10,7 +10,9 @@
 
   namespace :admin do
     resources :dashboard, only: [:index]
-    resources :companies, only: [:show, :destroy, :update]
+    resources :companies, only: [:show, :destroy, :update] do
+      resources :notes, only: [:index, :create, :destroy]
+    end
     resources :pdfs, only: [:create, :new]
     resource :tools_resources, only: [:edit]
   end
```
Comment (Lydias303) [inline config/routes.rb:18]: Do we really want routes for notes? I would think that this should just be on the company edit page.

File: app/views/admin/companies/show.html.erb [hunk 2]
```diff
@@ -4,6 +4,22 @@
...
+<div class="add_note">
+  <%= form_tag admin_company_notes_path(company_id: @company.id), method: :post do %>
+    <%= text_area_tag :note %>
+    <%= submit_tag "Add Note" %>
+  <% end %>
+</div>
+
 <%= button_to "Delete Company", admin_company_path(@company.id), method: :delete,
   data: { confirm: 'Are you sure you want to delete this company?
   This action cannot be undone.'} %>
```
Comment (Lydias303) [inline app/views/admin/companies/show.html.erb:30]: This looks like front end work.

File: app/models/company.rb [hunk 3]
```diff
@@ -1,6 +1,8 @@
 class Company < ActiveRecord::Base
   include ActiveModel::Dirty
 
+  has_many :notes
```
Comment (laurawhalin) [inline app/models/company.rb:4]: This wasn't fully specified, so it's interesting to see how you implemented notes. I was envisioning that each company had one note associated. Let's talk about implementation at the stand up tomorrow. 
Here are links to other products with the two options. I think has_one is more common, as far as I could see from a quick google search.
has_many notes: http://www.actpresentation.com/shots/quick_nav_list1.jpg
has_one notes: http://databasemelbourne.com/wp-content/uploads/2013/09/contactdetails.png

File: config/routes.rb (hunk 1 above)
Comment (laurawhalin) [inline config/routes.rb:18]: I am confused about this as well, but maybe because I was just expecting to see one notes section on each company show page. I think this could make sense if we want to maintain multiple notes for each company.

File: db/migrate/20150427223923_create_notes.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateNotes < ActiveRecord::Migration
+  def change
+    create_table :notes do |t|
+      t.string :body
+      t.references :company, index: true, foreign_key: true
```
Comment (laurawhalin) [inline db/migrate/20150427223923_create_notes.rb:5]: After adding indexes after the fact all morning long (during workshop), I really appreciate references now!
:sunglasses:

File: config/routes.rb (hunk 1 above)
Comment (oorion) [inline config/routes.rb:18]: Yes, because they could have many notes per company, I broke it out into its own model.  I agree that we should discuss this further.

File: spec/factories/notes.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+FactoryGirl.define do
+end
```
Comment (laurawhalin) [inline spec/factories/notes.rb]: Is this meant to be here? And be empty?

File: app/views/admin/companies/show.html.erb (hunk 2 above)
Comment (oorion) [inline app/views/admin/companies/show.html.erb:30]: I worked with Chris on this.

File: spec/factories/notes.rb (hunk 5 above)
Comment (oorion) [inline spec/factories/notes.rb]: It was created when we generated the model.  I can remove it.

File: spec/features/admin_company_management_spec.rb [hunk 6]
```diff
@@ -44,12 +44,26 @@ def login
...
+      end
+      click_link_or_button("Add Note")
+
+      within(".notes") do
+        expect(page).to have_content("test note")
+      end
+      expect(current_path).to eq(admin_company_path(companies.last.id))
+    end
   end
 end
```
Comment (laurawhalin) [inline spec/features/admin_company_management_spec.rb:102]: Just riffing off my previous comments, another option would be edit a pre-existing, empty note, perhaps?

File: spec/models/note_spec.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+require 'rails_helper'
+
+RSpec.describe Note, type: :model do
+end
```
Comment (laurawhalin) [inline spec/models/note_spec.rb:4]: I think if we stick with the has_many approach, it would be great to make sure we don't allow them to save empty notes.

File: spec/features/admin_company_management_spec.rb (hunk 6 above)
Comment (oorion) [inline spec/features/admin_company_management_spec.rb:102]: I agree.  However, the user story only involved adding notes.  We can add another user story, but I think that should be done in another PR.

File: spec/models/note_spec.rb (hunk 7 above)
Comment (oorion) [inline spec/models/note_spec.rb:4]: If that's something we want, let's add it as a chore.  It seems useful, but maybe not a priority.

File: config/routes.rb (hunk 1 above)
Comment (scottalexandra) [inline config/routes.rb:18]: I agree, As we discussed this morning, I think a notes controller is great, this gives us the ability for each indvidual admin to add notes, and the other admins can see who added the note.

File: spec/features/admin_company_management_spec.rb [hunk 8]
```diff
@@ -45,12 +45,30 @@ def login
...
+        fill_in "note", with: "test note"
+      end
+      click_link_or_button("Add Note")
+
+      within(".notes") do
+        expect(page).to have_content("test note")
+      end
+      expect(current_path).to eq(admin_company_path(companies.first.id))
+    end
+
```
Comment (scottalexandra) [inline spec/features/admin_company_management_spec.rb:71]: Nice Test! :fried_shrimp:
